addons/account/data/invoice_action_data.xml

```diff
@@ -62,6 +62,8 @@ from ${object.company_id.name}.
             <field name="auto_delete" eval="True"/>
             <field name="body_html"><![CDATA[<html>
                 <head></head>
+                % set record = ctx.get('record')
+                % set company = record and record.company_id or user.company_id
```

Comment:
I am not sure what happens if the record does not have company_id field defined ? False or error ?